core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -184,23 +165,27 @@ public BlockWorkerClient acquireWorkerClient(WorkerNetAddress address)
    */
   public BlockWorkerClient acquireLocalWorkerClient() {
     initializeLocalBlockWorkerClientPool();
-    if (mLocalBlockWorkerClientPool == null) {
+    if (mLocalBlockWorkerClientPoolMap.isEmpty()) {
       return null;
     }
-    return mLocalBlockWorkerClientPool.acquire();
+    // return any local worker
+    return mLocalBlockWorkerClientPoolMap.values().iterator().next().acquire();
```

Comment:
Would it make sense to select a random pool? Though I'm fine with this since in the normal case there should only be 1 pool.